data_structures/arrays/find_median_sorted_arrays.py

```diff
@@ -0,0 +1,83 @@
+"""
+https://atharayil.medium.com/median-of-two-sorted-arrays-day-36-python-fcbd2dbbb668
+"""
+
+
+def find_median_sorted_arrays(nums1: list[int], nums2: list[int]) -> float:
+    """
+    Finds the median of two sorted arrays.
+
+    :param nums1: The first sorted array.
+    :param nums2: The second sorted array.
+    :return: The median of the combined sorted arrays.
+    :raises ValueError: If both input arrays are empty.
+
+    >>> find_median_sorted_arrays([1, 3], [2])
```

Comment:
Test please for negative numbers, floating point numbers, letters, two empty arrays, arrays that are not sorted, etc?